trajectory_processing/src/iterative_time_parameterization.cpp

```diff
@@ -111,13 +112,18 @@ void IterativeParabolicTimeParameterization::applyVelocityConstraints(robot_traj
   {
     const robot_state::RobotStatePtr &curr_waypoint = rob_trajectory.getWayPointPtr(i);
     const robot_state::RobotStatePtr &next_waypoint = rob_trajectory.getWayPointPtr(i+1);
-    
+
+    double velocity_scaling_factor = 1.0;
+
+    if (max_velocity_scaling_factor > 0.0 && max_velocity_scaling_factor <= 1.0)
+      velocity_scaling_factor = max_velocity_scaling_factor;
```

Comment:
Arg, I just noticed this is inside a loop, which of course is unnecessary. Will move it to top of that loop.

Using an assert only is dangerous, as funny things will happen when a value of 0.0 gets used (I did that accidentally, and the trajectory is complete garbage). Allowing 0.0 to automatically get mapped to 1.0 silently makes behavior completely transparent for users that update, but do not update the MotionPlanRequests they use (value will default to 0.0, get mapped to 1.0 and everything works as before). Adding a logwarn would mean that suddenly everybody would get warnings about not setting the max_velocity_scaling_factor in their MotionPlanRequest to a valid value. If this is desired is a matter of taste/policy.